Dockerfile.dev

```diff
@@ -0,0 +1,19 @@
+FROM golang:1.21 as build
+
+ENV CGO_ENABLED=0
+
+WORKDIR /go/ingress-controller
+
+COPY go.mod go.sum /go/ingress-controller/
+RUN go mod download
+
+COPY . .
+RUN go build -o ./bin/ingress-controller ./cmd/caddy
+
+FROM gcr.io/distroless/static-debian11
```

Comment:
do we want distroless images when developping? It can be useful sometime to have some helper included in the image

Reply:
Main pros here is that the generated image can be "moved" faster between envs, but I get your point, do you feel  we need a full image or something like alpine is enough?

Reply:
Alpine should be enough and is really small (5MB)

Reply:
Moved to alpine 🙌